velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -1051,20 +1053,24 @@ uint32_t DateTimeFormatter::maxResultSize(const tz::TimeZone* timezone) const {
         size += std::max((int)token.pattern.minRepresentDigits, 9);
         break;
       case DateTimeFormatSpecifier::TIMEZONE:
-        if (timezone == nullptr) {
-          VELOX_USER_FAIL("Timezone unknown");
-        }
-        size += std::max(
-            token.pattern.minRepresentDigits, timezone->name().length());
+        VELOX_UNSUPPORTED(
+            "Date format specifier is not supported: {} ({})",
+            getSpecifierName(token.pattern.specifier),
+            token.pattern.minRepresentDigits);
+
         break;
       case DateTimeFormatSpecifier::TIMEZONE_OFFSET_ID:
-        if (token.pattern.minRepresentDigits != 2) {
-          VELOX_UNSUPPORTED(
-              "Date format specifier is not supported: {} ({})",
-              getSpecifierName(token.pattern.specifier),
-              token.pattern.minRepresentDigits);
+        if (token.pattern.minRepresentDigits == 1) {
+          size += 8;
+        } else if (token.pattern.minRepresentDigits == 2) {
+          size += 9;
+        } else {
+          if (timezone == nullptr) {
+            VELOX_USER_FAIL("Timezone unknown");
+          }
+          size += std::max(
```

Comment:
just curious, is this always true? like if `timezone->name().length()` is less than `token.pattern.minRepresentDigits` will the remaining characters be filled up with something? or does it not matter because we are only calculating the "`MAX`-ResultSize" here and the actual can be smaller?